### PR TITLE
Dataset Selection and Pathing

### PR DESCRIPTION
**Pull request by MattsonCam**
This pr updates dataset selection and pathing so that I can easily train on different datasets. The main branch will likely not be updated each time I train a new model or perform fine-tuning. Instead, I plan to log the runs in main and develop on different branches. However, I still want to make all of the models available in main. In a future prs I will modify the splitting based on the dataset selected and include additional models.

### CODE REVIEW

**Comment by MattsonCam** [comment]
The vanilla Unet models were in the repo before, but I accidentally removed them so I added them back in so that training can proceed with this model pipeline

File: datasets/dataset_00/utils/CropCacheBuilder.py [hunk 1]
```diff
@@ -143,46 +151,54 @@ def _build_filtered_profiles(data_dir: pathlib.Path) -> pd.DataFrame:
...
 
     required_cols = {
         "Metadata_Plate",
         "Metadata_Well",
         "Metadata_Site",
-        "Nuclei_AreaShape_BoundingBoxMinimum_X",
-        "Nuclei_AreaShape_BoundingBoxMaximum_X",
-        "Nuclei_AreaShape_BoundingBoxMinimum_Y",
-        "Nuclei_AreaShape_BoundingBoxMaximum_Y",
+        "Metadata_Nuclei_AreaShape_BoundingBoxMinimum_X",
```
Comment (wli51) [inline datasets/dataset_00/utils/CropCacheBuilder.py:202]: Purely out of interest, why does this new dataset treat bounding box columns as metadata as opposed to features, is this a cell profiler version thing? 

If that is the case, maybe adding some kind of fuzzy matching for column names would help to reduce the need to change code from dataset to dataset

File: models/UNetLayers.py [hunk 2]
```diff
@@ -0,0 +1,135 @@
...
+        in_channels: int,
+        out_channels: int,
+        kernel_size: int,
+        stride: int,
+        padding: int,
+        padding_mode: str = "zeros",
+        pooling: nn.Module = None,
+    ):
+        super().__init__()
+
```
Comment (wli51) [inline models/UNetLayers.py:68]: Consider documenting this layer as its implementation deviates from the conventional double convolution used in UNets (two consecutive spatial dimension preserving conv2d layers each followed by batch norm and relu).

Here I believe you have a customizable `x1` layer that can be configured to be spatial dimensional preserving by setting the appropriate stride and padding, plus a more rigid `xr` layer whose stride is separately configured for spatial down scaling convolution.

**Review by wli51 — APPROVED** [review]
LGTM

File: datasets/dataset_00/utils/CropCacheBuilder.py (hunk 1 above)
Comment (MattsonCam) [reply datasets/dataset_00/utils/CropCacheBuilder.py:202]: > Purely out of interest, why does this new dataset treat bounding box columns as metadata as opposed to features, is this a cell profiler version thing?

Jenna said it's "100%, just a sympton of cellprofiler evil naming lol". I think it makes more sense for them to be Metadata if they are used in machine learning pipelines, because then it is easy to remove them when training a model

Comment (MattsonCam) [reply datasets/dataset_00/utils/CropCacheBuilder.py:202]: > If that is the case, maybe adding some kind of fuzzy matching for column names would help to reduce the need to change code from dataset to dataset

Good idea, I've implement substring matching here

Comment (wli51) [reply datasets/dataset_00/utils/CropCacheBuilder.py:202]: 😈

File: models/UNetLayers.py (hunk 2 above)
Comment (MattsonCam) [reply models/UNetLayers.py:68]: Thanks for the suggestions, I have implement them

**Comment by MattsonCam** [comment]
Thanks @wli51 ! Merging now
